src/windows/simulate.rs

```diff
@@ -1,53 +1,71 @@
 use crate::rdev::{Button, EventType, SimulateError};
 use crate::windows::keycodes::code_from_key;
-use std::mem::{size_of, transmute, transmute_copy};
-use winapi::ctypes::c_int;
+use std::convert::TryFrom;
+use std::mem::size_of;
+use winapi::ctypes::{c_int, c_short};
+use winapi::shared::minwindef::{DWORD, UINT, WORD};
+use winapi::shared::ntdef::LONG;
 use winapi::um::winuser::{
-    GetSystemMetrics, SendInput, INPUT, INPUT_KEYBOARD, INPUT_MOUSE, KEYBDINPUT, KEYEVENTF_KEYUP,
-    LPINPUT, MOUSEEVENTF_ABSOLUTE, MOUSEEVENTF_HWHEEL, MOUSEEVENTF_LEFTDOWN, MOUSEEVENTF_LEFTUP,
-    MOUSEEVENTF_MIDDLEDOWN, MOUSEEVENTF_MIDDLEUP, MOUSEEVENTF_MOVE, MOUSEEVENTF_RIGHTDOWN,
-    MOUSEEVENTF_RIGHTUP, MOUSEEVENTF_WHEEL, MOUSEEVENTF_XDOWN, MOUSEEVENTF_XUP, MOUSEINPUT,
-    SM_CXVIRTUALSCREEN, SM_CYVIRTUALSCREEN, WHEEL_DELTA,
+    GetSystemMetrics, INPUT_u, SendInput, INPUT, INPUT_KEYBOARD, INPUT_MOUSE, KEYBDINPUT,
+    KEYEVENTF_KEYUP, MOUSEEVENTF_ABSOLUTE, MOUSEEVENTF_HWHEEL, MOUSEEVENTF_LEFTDOWN,
+    MOUSEEVENTF_LEFTUP, MOUSEEVENTF_MIDDLEDOWN, MOUSEEVENTF_MIDDLEUP, MOUSEEVENTF_MOVE,
+    MOUSEEVENTF_RIGHTDOWN, MOUSEEVENTF_RIGHTUP, MOUSEEVENTF_VIRTUALDESK, MOUSEEVENTF_WHEEL,
+    MOUSEEVENTF_XDOWN, MOUSEEVENTF_XUP, MOUSEINPUT, SM_CXVIRTUALSCREEN, SM_CYVIRTUALSCREEN,
+    WHEEL_DELTA,
 };
 /// Not defined in win32 but define here for clarity
-static KEYEVENTF_KEYDOWN: u32 = 0;
+static KEYEVENTF_KEYDOWN: DWORD = 0;
 
-fn mouse_event(flags: u32, data: u32, dx: i32, dy: i32) -> Result<(), SimulateError> {
-    let mut input = INPUT {
+fn sim_mouse_event(flags: DWORD, data: DWORD, dx: LONG, dy: LONG) -> Result<(), SimulateError> {
+    let mut union: INPUT_u = unsafe { std::mem::zeroed() };
+    let inner_union = unsafe { union.mi_mut() };
+    *inner_union = MOUSEINPUT {
+        dx,
+        dy,
+        mouseData: data,
+        dwFlags: flags,
+        time: 0,
+        dwExtraInfo: 0,
+    };
+    let mut input = [INPUT {
         type_: INPUT_MOUSE,
-        u: unsafe {
-            transmute(MOUSEINPUT {
-                dx,
-                dy,
-                mouseData: data,
-                dwFlags: flags,
-                time: 0,
-                dwExtraInfo: 0,
-            })
-        },
+        u: union,
+    }; 1];
+    let value = unsafe {
+        SendInput(
+            input.len() as UINT,
+            input.as_mut_ptr(),
+            size_of::<INPUT>() as c_int,
+        )
     };
-    let value = unsafe { SendInput(1, &mut input as LPINPUT, size_of::<INPUT>() as c_int) };
     if value != 1 {
         Err(SimulateError)
     } else {
         Ok(())
     }
 }
 
-fn keyboard_event(flags: u32, vk: u16, scan: u16) -> Result<(), SimulateError> {
-    let mut input = INPUT {
+fn sim_keyboard_event(flags: DWORD, vk: WORD, scan: WORD) -> Result<(), SimulateError> {
+    let mut union: INPUT_u = unsafe { std::mem::zeroed() };
+    let inner_union = unsafe { union.ki_mut() };
+    *inner_union = KEYBDINPUT {
```

Comment:
while more verbose, this is how the winapi crate recommends you instantiate unions.